pilot/pkg/model/gateway.go

```diff
@@ -286,12 +288,24 @@ func mergeGateways(gateways []gatewayWithInstances, proxy *Proxy, ps *PushContex
 							ms.RouteName = routeName
 							ms.Servers = append(ms.Servers, s)
 						} else {
-							// Merge this to current known port with same bind.
+							// Same bind, merge with existing server
 							ms := mergedServers[current]
 							ms.Servers = append(ms.Servers, s)
 						}
 						serversByRouteName[routeName] = append(serversByRouteName[routeName], s)
 					} else {
+						// No existing plaintext server on this port
+						// Check if current server is plaintext and should be handled differently
+						if !gateway.IsTLSServer(s) {
+							// This is a plain text server, add it to the map
+							plainTextServers[resolvedPort] = serverPort
+							if gateway.IsHTTPServer(s) {
+								serversByRouteName[routeName] = []*networking.Server{s}
+							}
+							mergedServers[serverPort] = &MergedServers{Servers: []*networking.Server{s}, RouteName: routeName}
+							serverPorts = append(serverPorts, serverPort)
```

Comment:
This is duplicate with original L357-L380 And missing quic process. 

So i would suggest abstract a function for them